charts/consul/templates/client-snapshot-agent-deployment.yaml

```diff
@@ -29,10 +30,16 @@ spec:
       annotations:
         "consul.hashicorp.com/connect-inject": "false"
         {{- if .Values.global.secretsBackend.vault.enabled }}
+        {{- if and .Values.global.tls.enabled .Values.client.snapshotAgent.configSecret.secretName }}
+        "vault.hashicorp.com/role": {{ .Values.global.secretsBackend.vault.consulCARole }},{{ .Values.global.secretsBackend.vault.consulSnapshotAgentRole }}
```

Comment:
I think vault can only have one role, so in this case it would be the `consulSnapshotAgentRole`. 

Note that the user would not need to allow access to the CA for the `consulSnapshotAgentRole` if they have configured it like we recommend in our docs allowing all service accounts (`*`) to bind to it : https://www.consul.io/docs/k8s/installation/vault/server-tls. The CA role is really only needed when you need access to the CA only because vault agent will not start without a role provided. Hope it makes sense!